code/game/objects/structures/maintenance.dm

```diff
@@ -0,0 +1,99 @@
+/** This structure acts as a source of moisture loving cell lines,
+as well as a location where a hidden item can somtimes be retrieved
+at the cost of risking a vicious bite.**/
+/obj/structure/moisture_trap
+	name = "moisture trap"
+	desc = "A device installed in order to control moisture in poorly ventilated areas.\nThe stagnant water inside basin seems to produce serious biofouling issues when improperly maintained.\nThis unit in particular seems to be teeming with life!\nWho thought mother Gaia could assert herself so vigoriously in this sterile and desolate place?"
+	icon_state = "moisture_trap"
+	anchored = TRUE
+	density = FALSE
+	///This var stores the hidden item that might be able to be retrieved from the trap
+	var/obj/item/hidden_item
+	///This var determines if there is a chance to recieve a bite when sticking your hand into the water.
+	var/critter_infested = TRUE
+	var/list/loot = list(
+					/obj/item/reagent_containers/food/snacks/meat/slab/human/mutant/skeleton = 35,
+					/obj/item/reagent_containers/food/snacks/meat/slab/human/mutant/zombie = 15,
+					/obj/item/trash/can = 15,
+					/obj/item/clothing/head/helmet/skull = 10,
+					/obj/item/restraints/handcuffs = 4,
+					/obj/item/restraints/handcuffs/cable/red = 1,
+					/obj/item/restraints/handcuffs/cable/blue = 1,
+					/obj/item/restraints/handcuffs/cable/green = 1,
+					/obj/item/restraints/handcuffs/cable/pink = 1,
+					/obj/item/restraints/handcuffs/alien = 2,
+					/obj/item/coin/bananium = 9,
+					/obj/item/kitchen/knife/butcher = 5,
+					/obj/item/coin/mythril = 1) //the loot table isn't that great and should probably be improved and expanded later.
+
+
+/obj/structure/moisture_trap/Initialize()
+	. = ..()
+	if(prob(40))
+		critter_infested = FALSE
+	if(prob(75))
+		var/picked_item = pickweight(loot)
+		hidden_item = new picked_item(src)
+	AddElement(/datum/element/swabable, CELL_LINE_TABLE_MOIST, CELL_VIRUS_TABLE_GENERIC, rand(2,4), 20)

```

Comment:
```suggestion
	if(prob(75))
		var/picked_item = pickweight(loot)
		hidden_item = new picked_item(src)
	loot = null
	AddElement(/datum/element/swabable, CELL_LINE_TABLE_MOIST, CELL_VIRUS_TABLE_GENERIC, rand(2,4), 20)
```
If you don't use the list anymore, clear it.

Reply:
??

Reply:
oh